### DISCUSSION TITLE
How to get a list of pages?

### DISCUSSION THREAD

**Discussion by luminarious** [nuxt/nuxt#17900] [Questions] [answered]
Hey! Under [pages](https://v3.nuxtjs.org/docs/directory-structure/pages), there is information about individual pages, but I couldn't find any hints as to how I would get a list of pages?

Typical use case is showing a bunch of links on the homepage:
```js
const articles = usePages('~/articles/**/*.vue')

// Or if using the router data then maybe
const articles = usePages('/articles/**/*')
```
I should hope to get an array of page data, like `[Page{ route, meta }]`

**Comment by danielroe** [comment, marked as answer]
Try:

```js
const routes = useRouter().options.routes
```

**Reply by luminarious** [reply]
Ok, thank you. I'll try to make a composable based on this.
